pkg/config/constants/constants.go

```diff
@@ -199,6 +198,8 @@ const (
 
 	// AmbientUseWaypoint is the annotation used to specify which waypoint should be used for a given pod, service, etc...
 	AmbientUseWaypoint = "istio.io/use-waypoint"
+	// AmbientWaypointForTrafficType is the annotation used to specify which traffic should go through the intended waypoint.
+	AmbientWaypointForTrafficType = "istio.io/waypoint-for"
```

Comment:
can you add more comments what type is valid, and which resource can this annotation be applied on?

Reply:
Will do